evaldocsloader/loader_fetch.py

```diff
@@ -182,6 +186,57 @@ def _edit_user_docs(self, doc: mistletoe.Document) -> mistletoe.Document:
         response_areas_content = format_response_areas(supported_response_types)
         doc.children.insert(heading + 1, mistletoe.block_token.Paragraph([response_areas_content]))
 
+        # Insert a section at the end with examples auto-generated from tests, if a tests file exists
+        if self._test_file:
+            logger.info(f"Test file found for {self._repo.name}, generating examples")
+            # Append the content to the end of the file
+            doc.children.append(mistletoe.block_token.Heading((2, "Examples from Integration Tests", None)))
+            
+            # The table header is the same for all tests
+            table_header = [
+                "|Response|Answer|Correct?|",
+                "|-|-|-|",
+            ]
+            
+            for group in self._test_file.groups:
+                doc.children.append(mistletoe.block_token.Heading((3, group.title, None)))
+                for test in group.tests:
+                    if test.exclude_from_docs:
+                        continue
+                    
+                    doc.children.append(mistletoe.block_token.Paragraph([test.desc]))
+                    doc.children.append(mistletoe.markdown_renderer.BlankLine({}))
+
+                    # A buffer that allows tables for multiple sub-tests to be combined
+                    table_lines = []
+                    
+                    # Sub tests have the same answer and parameters as a test, but a different response value
+                    for sub_test in test.sub_tests:
+                        if sub_test.exclude_from_docs:
+                            continue
+                        
+                        response = sanitise_response(sub_test.response)
+                        answer = sanitise_response(test.answer)
```

Comment:
Is sanitation necessary, as both `response` and `answer` are anyways wrapped in a inline-code "`"?

Reply:
MkDocs renders `|` characters between backticks correctly, but mistletoe still interprets them as column separators and adds unnecessary columns.
Here's what the raw markdown looks like;
With sanitisation:
```markdown
| Response                   | Answer  | Correct? |
| -------------------------- | ------- | -------- |
| `~(~(~A | ~B) | ~(A | B))` | `A ^ B` | ✓        |
```
Without sanitisation:
```markdown
| Response | Answer | Correct? |      |         |     |
| -------- | ------ | -------- | ---- | ------- | --- |
| `~(~(~A  | ~B)    | ~(A      | B))` | `A ^ B` | ✓   |
```

This adds unnecessary whitespace to the right:

![Screenshot 2024-09-06 110315](https://github.com/user-attachments/assets/f857ea49-5d20-4d5f-8228-8bee0fdc85ed)

When it looks like this with sanitisation:

![Screenshot 2024-09-06 110442](https://github.com/user-attachments/assets/4bb52dd3-b55a-4c54-8def-4cb5ba64b784)

Reply:
Hilarious, ok perfectly fine wanted to understand!